packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -153,17 +153,21 @@ class SectionHeader extends React.Component {
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     /* eslint-disable jsx-a11y/no-static-element-interactions */
 
-    const buttonAttributes = (onClick) ? { role: 'button', 'aria-expanded': isOpen, 'aria-label': headerText } : undefined;
+    const elementAttributes = (onClick && accordionIcon) ? {
+      type: 'button', tabIndex: '-1', 'aria-expanded': isOpen, 'aria-label': headerText,
+    } : { tabIndex: '-1' };
```

Comment:
```suggestion
const buttonAttributes = (onClick && accordionIcon) ? { role: 'button', 'aria-expanded': isOpen, 'aria-label': headerText } : undefined;
```
let's keep this attributes as button since these attributes are added only for button.

Reply:
Updated